pm_yaml_config/pm_yaml_config.xml

```diff
@@ -0,0 +1,126 @@
+        #else if str($calculator_cond.calculator)=="dftb+": 
+            printf "dftb_set: $calculator_cond.dftb_set \ndftb_pbc: $calculator_cond.dftb_pbc \n">>outputx.yaml &&
+        #else if str($calculator_cond.calculator)=="all": 
+            printf "uep_gw_factor: $calculator_cond.uep_gw_factor \nuep_chden: $calculator_cond.uep_chden \ncastep_command: $calculator_cond.castep_command \ncastep_param: $calculator_cond.castep_param \nmu_symbol: $calculator_cond.mu_symbol \ndftb_set: $calculator_cond.dftb_set \ndftb_optionals: $calculator_cond.dftb_optionals \ndftb_pbc: $calculator_cond.dftb_pbc \n">>outputx.yaml &&
+        #end if 
+        ln -s outputx.yaml output.yaml
+    ]]></command>
+    <inputs>
+        <param type="float" name="poisson_r" value="0.8" label="poisson radius" help=" Poisson sphere radius to use for random generation. No two starting muon positions will be closer than this distance. Smaller values make for bigger structure sets."/>
+        <param type="text" name="struct_name" label="name" help="name to call the folder for containing each structure. This name will be postfixed with a unique number, e.g. struct_001."/>
+        <param type="select" name="charged" label="charged muon" value="true" help="if True, the muon will be considered charged instead of a muonium with an accompanying electron. Must be True for UEP calculations.">
+            <option value="true">true</option>
+            <option value="false">false</option>
+        </param>
+        <param type="integer" name="geom_steps" value="300" label="geometry optimization steps" help="maximum number of geometry optimisation steps."/>
+        <param type="float" name="vdw_scale" value="0.25" label="Van der Waal scale" help="Van der Waals scaling factor to use when generating muon sites to avoid existing atoms. Smaller values will allow muons to get closer to the other ions."/>
+        <param type="text" name="out_folder" value="muon-airss-out" label="output folder name" help="name to call the output folder used to store the input files that the script generates."/>
+        <param type="float" name="geom_force_tol" value="0.05" label="geometry optimization tolerance" help="tolerance on geometry optimisation in units of eV/Armstrong."/>
+        <conditional name="calculator_cond">
+            <param type="select" name="calculator" value="uep" help="calculator to generate structure files for. Must be a single word or a comma seperated list of values. Currently supported calculators are CASTEP, DFTB+ and UEP. Can also pass all as an option to generate files for all calculators.">
+                <option value="uep">uep</option>
+                <option value="castep">castep</option>
+                <option value="dftb+">dftb</option>
+                <option value="all">all</option>
+            </param>
+            <when value="uep">
```

Comment:
For all of these calculator-specific parameters, we could add the calculator name at the start of the label to make it clear what they apply to (useful for the "all" case). Also helps to denote where the calculator-specific params start and end if we can't indent them or anything.

Reply:
That's much better. 

Is it possible to do e.g. <when value="uep" or value="all"> to avoid duplicating all the parameters again under the "all" section? I guess you may have tried that already